src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -61,32 +71,91 @@ internal TagHelperOutput(string tagName)
         /// Content that precedes the HTML element.
         /// </summary>
         /// <remarks>Value is rendered before the HTML element.</remarks>
-        public TagHelperContent PreElement { get; } = new DefaultTagHelperContent();
+        public TagHelperContent PreElement
+        {
+            get
+            {
+                if (_preElement == null)
+                {
+                    _preElement = new DefaultTagHelperContent();
+                }
+
+                return _preElement;
+            }
+        }
 
         /// <summary>
         /// The HTML element's pre content.
         /// </summary>
         /// <remarks>Value is prepended to the <see cref="ITagHelper"/>'s final output.</remarks>
-        public TagHelperContent PreContent { get; } = new DefaultTagHelperContent();
+        public TagHelperContent PreContent
+        {
+            get
+            {
+                if (_preContent == null)
+                {
+                    _preContent = new DefaultTagHelperContent();
+                }
+
+                return _preContent;
+            }
+        }
 
         /// <summary>
-        /// The HTML element's main content.
+        /// Get or set the HTML element's main content.
         /// </summary>
         /// <remarks>Value occurs in the <see cref="ITagHelper"/>'s final output after <see cref="PreContent"/> and
         /// before <see cref="PostContent"/></remarks>
-        public TagHelperContent Content { get; } = new DefaultTagHelperContent();
+        public TagHelperContent Content
+        {
+            get
+            {
+                if (_content == null)
+                {
+                    _content = new DefaultTagHelperContent();
+                }
+
+                return _content;
+            }
+            set
+            {
+                _content = value ?? _content;
```

Comment:
Should it be an exception to set this to `null`? Generally doing something like this gives me the heebie-jeebies.